### PR TITLE
Add web doc for bare-metal codegen/run

### PR DESCRIPTION
**Pull request by hcindyl**
Part of https://github.com/google/iree/issues/6327

### CODE REVIEW

File: docs/website/docs/deployment-configurations/bare-metal.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# Run on a Bare-Metal Platform
+
+IREE supports CPU model execution on a bare-metal platform. That is, a platform
+without operating system support and the executable runs on the machine mode.
```
Comment (ScottTodd) [inline docs/website/docs/deployment-configurations/bare-metal.md:4]: can you reword or clarify "and the executable runs on the machine mode"?

File: docs/website/docs/deployment-configurations/bare-metal.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+The IREE compiler follows the path to generate the code runs on the
+[dynamic library CPU HAL driver](https://google.github.io/iree/deployment-configurations/cpu-dylib/)
+and a subset of runtime library can be used to execute the model.
```
Comment (ScottTodd) [inline docs/website/docs/deployment-configurations/bare-metal.md:7]: Let's break this apart a bit:

```suggestion
Bare-metal deployment typically uses IREE's LLVM compiler target much like the 
[CPU - Dylib](https://google.github.io/iree/deployment-configurations/cpu-dylib/)
configuration, but using a limited subset of IREE's CPU HAL driver at runtime to
load and execute compiled programs.
```

File: docs/website/docs/deployment-configurations/bare-metal.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+Out-of-tree bare-metal platform tools and source code for the system should be
+ready, such as
+
+* Compilation toolchain
+* Platform linker script
+* Firmware libraries
+
+Please follow the
+[instructions](https://google.github.io/iree/deployment-configurations/cpu-dylib/#get-compiler-for-cpu-native-instructions)
```
Comment (ScottTodd) [inline docs/website/docs/deployment-configurations/bare-metal.md:19]: Does a relative path work here?

```suggestion
[instructions](./cpu-dylib.md#get-compiler-for-cpu-native-instructions)
```

Seems like it should: https://mkdocs.readthedocs.io/en/0.10/user-guide/writing-your-docs/#linking-documents

That would have the advantage of working when developing locally or publishing to a different domain (such as your own github pages).

File: docs/website/docs/deployment-configurations/bare-metal.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+* `iree-llvm-target-triple`: Use the `<arch>-pc-linux-elf` LLVM target triple so
+the artifact has a fixed ABI to be rendered by the
+[elf_module library](https://github.com/google/iree/tree/main/iree/hal/local/elf)
+* `iree-llvm-debug-symbols=false`: To reduce the artifact size
+
+See [generate.sh](https://github.com/google/iree/blob/main/iree/hal/local/elf/testdata/generate.sh)
+for example command-line instructions of some common architectures
+
+You can replace the MLIR file with the other MLIR model files, following the
+[instructions](https://google.github.io/iree/deployment-configurations/cpu-dylib/#compile-the-model)
```
Comment (ScottTodd) [inline docs/website/docs/deployment-configurations/bare-metal.md:56]: Same here - relative links are preferred

```suggestion
[instructions](./cpu-dylib.md#compile-the-model)
```

File: docs/website/docs/deployment-configurations/bare-metal.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+!!! todo
+    add the static library support
+
+## Build bare-metal runtime from the source (CMake only)
```
Comment (ScottTodd) [inline docs/website/docs/deployment-configurations/bare-metal.md:61]: I'd just drop "(Cmake only)" from here. I've intentionally omitted all mention of Bazel from the website :P

```suggestion
## Build bare-metal runtime from the source
```

File: docs/website/docs/deployment-configurations/bare-metal.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+* `set(IREE_ENABLE_MLIR OFF)`: Disable LLVM/MLIR dependencis for the runtime
+library
+* `set(CMAKE_SYSTEM_NAME Generic)`: Let CMake to generate the runtime without
+targeting a specific operating system
+* `set(IREE_BINDINGS_TFLITE OFF)`: Disable the TFLite binding support
+* `set(IREE_ENABLE_THREADINGS OFF)`: Disable multi-thread library support
+* `set(IREE_HAL_DRIVERS_TO_BUILD "Dylib;VMVX")`: Build the runtime HAL drivers
+for the dynamic library and VMVX backends.
+* `set(IREE_BUILD_TESTS OFF)`: IREE test tools assumes the multi-thread system
+support
```
Comment (ScottTodd) [inline docs/website/docs/deployment-configurations/bare-metal.md:80]: grammar/spelling nits and some rewording suggestions

```suggestion
* `set(IREE_BUILD_COMPILER OFF)`: Build IREE runtime only
* `set(IREE_ENABLE_MLIR OFF)`: Disable LLVM/MLIR dependencies for the runtime
library
* `set(CMAKE_SYSTEM_NAME Generic)`: Tell CMake to skip targeting a specific operating system
* `set(IREE_BINDINGS_TFLITE OFF)`: Disable the TFLite binding support
* `set(IREE_ENABLE_THREADING OFF)`: Disable multi-thread library support
* `set(IREE_HAL_DRIVERS_TO_BUILD "Dylib;VMVX")`: Build only the dynamic library and VMVX runtime HAL drivers
* `set(IREE_BUILD_TESTS OFF)`: Disable tests until IREE supports running them on bare-metal platforms
support
```

**Review by ScottTodd — CHANGES_REQUESTED** [review]

File: docs/website/docs/deployment-configurations/bare-metal.md (hunk 3 above)
Comment (hcindyl) [reply docs/website/docs/deployment-configurations/bare-metal.md]: Done. Thanks for the tip!

File: docs/website/docs/deployment-configurations/bare-metal.md (hunk 4 above)
Comment (hcindyl) [reply docs/website/docs/deployment-configurations/bare-metal.md]: Done.

File: docs/website/docs/deployment-configurations/bare-metal.md (hunk 5 above)
Comment (hcindyl) [reply docs/website/docs/deployment-configurations/bare-metal.md]: Done.

File: docs/website/docs/deployment-configurations/bare-metal.md (hunk 6 above)
Comment (hcindyl) [reply docs/website/docs/deployment-configurations/bare-metal.md]: Done.

File: docs/website/docs/deployment-configurations/bare-metal.md (hunk 2 above)
Comment (hcindyl) [reply docs/website/docs/deployment-configurations/bare-metal.md]: Done.

File: docs/website/docs/deployment-configurations/bare-metal.md (hunk 1 above)
Comment (hcindyl) [reply docs/website/docs/deployment-configurations/bare-metal.md]: Reword the paragraph.

File: docs/website/mkdocs.yml [hunk 7]
```diff
@@ -100,6 +100,7 @@ nav:
       - TensorFlow Lite: 'ml-frameworks/tensorflow-lite.md'
       - JAX: 'ml-frameworks/jax.md'
   - 'Deployment configurations':
+      - CPU - Bare-Metal: 'deployment-configurations/bare-metal.md'
       - CPU - Dylib: 'deployment-configurations/cpu-dylib.md'
```
Comment (ScottTodd) [inline docs/website/mkdocs.yml:104]: The ordering used here will be reflected in the nav:

image

Let's switch these so dylib is first, since the bare metal doc references to the dylib doc.

```suggestion
      - CPU - Dylib: 'deployment-configurations/cpu-dylib.md'
      - CPU - Bare-Metal: 'deployment-configurations/bare-metal.md'
```

**Review by ScottTodd — APPROVED** [review]

File: docs/website/mkdocs.yml (hunk 7 above)
Comment (hcindyl) [reply docs/website/mkdocs.yml]: Done. Was thinking about the alphabetical order, but definitely more sense with dyllib up front.
